Python/dtoa.c

```diff
@@ -361,9 +351,10 @@ Balloc(int k)
     int x;
     Bigint *rv;
     unsigned int len;
-
-    if (k <= Kmax && (rv = freelist[k]))
-        freelist[k] = rv->next;
+    PyInterpreterState *is = PyInterpreterState_Get();
```

Comment:
I would prefer to add a helper function get_freelist().

I suggest to use _PyInterpreterState_GET() which is faster than PyInterpreterState_Get().